### ISSUE TITLE
Feature Request - support logging of structured data to journald

### ISSUE THREAD

**Issue by peterhoeg** [sensu/sensu#1816]
sensu is currently sending its structured JSON logging data to stdout/stderr where it is picked up by journald and stored as a "text blob" making it unnecessarily difficult to parse.

It would be fantastic if the logging data could be sent properly to journald so that as an example "level" is picked up correctly and we could use the features of journalctl to look at and query the logging data.

As an example, currently the following are all logged as normal log messages, while they clearly are not:

```
Feb 28 14:53:39 app3 sensu-api[633]: {"timestamp":"2018-02-28T14:53:39.384936+0800","level":"error","message":"[amqp] Detected TCP connection failure: Errno::ECONNREFUSED"}
Feb 28 14:53:43 app3 uchiwa[526]: {"timestamp":"2018-02-28T14:53:43.964653208+08:00","level":"warn","message":"GET http://127.0.0.1:4567/stashes returned: 500 Internal Server Error"}
```

If the logging data was structured, we could do:

```
journalctl -u sensu.slice -p err
```

And only get the error messages.

## Your Environment

* Sensu version used: 1.2.1

**Comment by mbbroberg** [comment]
Very cool suggestion @peterhoeg! The core team will review this as part of our weekly triage (that happens to be tomorrow - good timing 😄). 

Could you provide a bit more detail on the current challenges you face when parsing errors? It will be helpful as we prioritize backlog. Thanks!

**Comment by peterhoeg** [comment]
Sure - there are a few different use-cases:

1) When setting up and/or troubleshooting sensu, you typically set the ```LOG_LEVEL``` to ```info``` or ```debug``` which produces a very large quantity of messages. Finding errors and warnings is difficult. Obviously not "travelling sales man difficult" but if would be far easier if we could simply query using the levels that journald supports.

2) Currently looking for any kind of specific output means grep. 
If as an example the "file" key/value here was stored in its own field in the following journal entry, you would be able to query based on that and easily find any mentioning of the file:

```
Feb 28 14:53:39 app3 sensu-api[633]: {"timestamp":"2018-02-28T14:53:38.949637+0800","level":"warn","message":"loading config file","file":"/etc/sensu/conf.d/client.json"}
```

So you could do this:

``` shell
  journalctl -u sensu-api _file=/etc/sensu/conf.d/client.json
```

3) journald supports tagging messages with other kinds of meta data so as an example, using a message id (a uuid basically) would allow you to find all the messages of a certain type without having it dumped on you when looking at the output

4) You could also dump your timestamp since all log entries are already tagged for you. Then the user can choose the resolution to see.

**Comment by peterhoeg** [comment]
One more thing, with structured logging data, the sensu-plugins-logs plugin will be able to check sensu as well using logging levels:

https://github.com/sensu-plugins/sensu-plugins-logs/blob/master/bin/check-journal.rb

**Comment by cwjohnston** [comment]
In talking to the engineering team it seems we are already supporting this use case in http://github.com/sensu/sensu-go (aka Sensu 2.0). I think we're open to supporting this in Sensu 1.x should someone in the community be willing to work on the implementation in http://github.com/sensu/sensu-logger.

**Comment by cwjohnston** [comment]
Closing this as we aren't planning to address it in Sensu 1.x and it is fixed in Sensu 2.0
